cmd/lotus-wallet/main.go

```diff
@@ -167,19 +204,43 @@ var runCmd = &cli.Command{
 			w = &LoggedWallet{under: w}
 		}
 
+		rpcApi := metrics.MetricedWalletAPI(w)
+		if !cctx.Bool("disable-auth") {
+			rpcApi = api.PermissionedWalletAPI(rpcApi)
+		}
+
 		rpcServer := jsonrpc.NewServer()
-		rpcServer.Register("Filecoin", metrics.MetricedWalletAPI(w))
+		rpcServer.Register("Filecoin", rpcApi)
 
 		mux.Handle("/rpc/v0", rpcServer)
 		mux.PathPrefix("/").Handler(http.DefaultServeMux) // pprof
 
-		/*ah := &auth.Handler{
-			Verify: nodeApi.AuthVerify,
-			Next:   mux.ServeHTTP,
-		}*/
+		var handler http.Handler = mux
+
+		if !cctx.Bool("disable-auth") {
+			authKey, err := modules.APISecret(ks, lr)
+			if err != nil {
+				return xerrors.Errorf("setting up api secret: %w", err)
+			}
+
+			authVerify := func(ctx context.Context, token string) ([]auth.Permission, error) {
+				var payload jwtPayload
+				if _, err := jwt.Verify([]byte(token), (*jwt.HMACSHA)(authKey), &payload); err != nil {
+					return nil, xerrors.Errorf("JWT Verification failed: %w", err)
```

Comment:
@magik6k Could we add a log entry here or is the returned error going to be logged eventually?